src/utils/temporal.cpp

```diff
@@ -79,36 +79,14 @@ Date::Date(const DateParameters &date_parameters) {
   day = date_parameters.day;
 }
 
-namespace {
-tm GetUtcFromSystemClockOrThrow() {
-  namespace chrono = std::chrono;
-  const auto today = chrono::system_clock::to_time_t(chrono::system_clock::now());
-  tm utc_today;
-  if (!gmtime_r(&today, &utc_today)) {
-    throw temporal::InvalidArgumentException("Can't access clock's UTC time");
-  }
-  return utc_today;
-}
-
-int64_t TMYearToUtcYear(int year) { return year + 1900; }
+Date UtcToday() { return UtcLocalDateTime().date; }
 
-int64_t TMMonthToUtcMonth(int month) { return month + 1; }
-}  // namespace
-
-Date UtcToday() {
-  const auto utc_today = GetUtcFromSystemClockOrThrow();
-  return Date({TMYearToUtcYear(utc_today.tm_year), TMMonthToUtcMonth(utc_today.tm_mon), utc_today.tm_mday});
-}
-
-LocalTime UtcLocalTime() {
-  const auto utc_today = GetUtcFromSystemClockOrThrow();
-  return LocalTime({utc_today.tm_hour, utc_today.tm_min, utc_today.tm_sec});
-}
+LocalTime UtcLocalTime() { return UtcLocalDateTime().local_time; }
 
 LocalDateTime UtcLocalDateTime() {
-  const auto utc_today = GetUtcFromSystemClockOrThrow();
-  return LocalDateTime({TMYearToUtcYear(utc_today.tm_year), TMMonthToUtcMonth(utc_today.tm_mon), utc_today.tm_mday},
-                       {utc_today.tm_hour, utc_today.tm_min, utc_today.tm_sec});
+  namespace chrono = std::chrono;
+  auto ts = chrono::time_point_cast<chrono::microseconds>(chrono::system_clock::now());
+  return LocalDateTime(ts.time_since_epoch().count());
```

Comment:
Will this return UTC time? I guess no, but you opted from chrono to be consistent with the timestamp thingy. And that is fine, however I think these function should be renamed, because in this case their name is misleading.

Reply:
You are right and I will very happily rename it!